packages/theme/src/theme.ts

```diff
@@ -261,6 +265,7 @@ export class Theme extends HTMLElement implements ThemeKindProvider {
             Theme.defaultFragments.add(name);
         }
         fragmentMap.set(name, { name, styles });
+        Theme.instances.map((instance) => instance.adoptStyles());
```

Comment:
Should we block this after `this.hasAdoptedStyles === true`?

Reply:
If the theme requirements (color / scale) change dynamically, and the theme fragments are still being late loaded, would we need to re-adopt?

Reply:
Hrm...I'll have to think about that a little. Currently, in `attributeChangedCallback` we re-adopt...if that were to be associated with further lazy loading, you _could_ reset `this.hasAdoptedStyles = false;` until the theme was satisfied again...thoughts?